core/arch/arm/mm/core_mmu.c

```diff
@@ -994,6 +994,101 @@ static void set_pg_region(struct core_mmu_table_info *dir_info,
 	}
 }
 
+TEE_Result core_mmu_map_pages(vaddr_t vstart, paddr_t *pages, size_t num_pages,
+			      enum teecore_memtypes memtype)
+{
+	TEE_Result ret;
+	struct core_mmu_table_info tbl_info;
+	struct tee_mmap_region *mm;
+	unsigned int idx;
+	uint32_t old_attr;
+	vaddr_t vaddr = vstart;
+	size_t i;
+
+	if (vaddr & SMALL_PAGE_MASK)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	mm = find_map_by_va((void *)vaddr);
+
+	if (!mm)
+		panic("VA does not belong to any known mm region");
```

Comment:
I'm all for avoiding `assert()` and `panic()` when nothing is broken (yet). I mean, if the caller is asking for something stupid, the function detects the stupidity and refuses to perform the action. It is the caller's responsibility to handle the error properly (as for many other error cases).

On the other hand, `panic()` and `assert()` should be used when something is _already_ in an inconsistent state, and proceeding anyway could cause unexpected behavior later. That is really the key criteria, I think. The two functions are pretty much interchangeable, it's just that one may be more convenient than the other (depending on whether you want to print a specific message or not, or you want the condition to appear in the log...).